src/bootstrap/bin/main.rs

```diff
@@ -15,10 +15,12 @@ fn main() {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
 
+    #[cfg(all(any(unix, windows), not(target_os = "solaris")))]
```

Comment:
Shouldn't this check only if not solaris?

Reply:
We currently lock only on unix and windows, but not solaris. There may be other platforms that are used as host bootstrap that aren't unix-like or windows, and we don't support locking for them either.